### ISSUE TITLE
How to exclude files/directories from find-file using ivy

### ISSUE THREAD

**Issue by viocost** [hlissner/doom-emacs#3061]
> Please visit our Discord server to ask how-to and workflow questions:
> https://discord.gg/qvGgnVx -- post on Doom's issue tracker as a last resort.

**What are you trying to achieve?**
When I hit SPACE SPACE to find file, I want to be able to exclude certain directories from search

**What have you tried?**
...
I tried adding .projectile to the root of my project with
```
- /dir
- /dir2
```
Also per ivy documentation tried adding .ignore to the root with
```
/dir
/dir2
 ```
also tried in my config.el 
```
 (setq projectile-globally-ignored-directories '("dir" "dir2")) 
```

**Additional information**
- Do you have screenshots/casts of this workflow in another editor or Emacs
  distro? (If possible, include names of specific commands or functions from
  other editors/emacs distros)
- Do you have additional material or links to resources that could help clarify
  what you are trying to do?

**System information**
<details><pre>
Include the output of `M-x doom/info` or `~/.emacs.d/bin/doom info` here.
```
emacs   version    26.3
        features   XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND GPM DBUS GSETTINGS GLIB NOTIFY ACL GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 XDBE XIM MODULES THREADS LIBSYSTEMD LCMS2
        build      Aug 30, 2019
        buildopts  (--prefix=/usr --sysconfdir=/etc --libexecdir=/usr/lib --localstatedir=/var --with-x-toolkit=gtk3 --with-xft --with-modules 'CFLAGS=-march=x86-64 -mtune=generic -O2 -pipe -fno-plt' CPPFLAGS=-D_FORTIFY_SOURCE=2 LDFLAGS=-Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now)
        windowsys  x
        daemonp    server-running
doom    version    2.0.9
        build      HEAD -> develop, origin/develop, origin/HEAD f9eb6a3b6 2020-02-11 19:24:08 -0500
        dir        ~/.doom.d/
system  type       gnu/linux
        config     x86_64-pc-linux-gnu
        shell      /usr/bin/zsh
        uname      Linux 5.6.7-arch1-1 #1 SMP PREEMPT Thu, 23 Apr 2020 09:13:56 +0000 x86_64
        path       (~/projects/islands/chat/node_modules/.bin/ ~/.nvm/versions/node/v12.16.1/bin /usr/local/sbin /usr/local/bin /usr/bin /usr/lib/jvm/default/bin /usr/bin/site_perl /usr/bin/vendor_perl /usr/bin/core_perl /usr/lib/emacs/26.3/x86_64-pc-linux-gnu/)
config  envfile    envvar-file
        elc-files  0
        modules    (:completion company ivy :ui deft doom doom-dashboard hl-todo indent-guides modeline nav-flash ophints (popup +all +defaults) treemacs vc-gutter vi-tilde-fringe window-select workspaces :editor (evil +everywhere) file-templates fold multiple-cursors rotate-text snippets :emacs dired electric ibuffer vc :term eshell term :checkers syntax spell :tools debugger docker (eval +overlay) (lookup +docsets) magit :lang data emacs-lisp javascript markdown (org +dragndrop +jupyter +present) python sh web :email (mu4e +gmail) :app irc :config (default +bindings +smartparens))
        packages   ((treemacs-magit) (treemacs-icons-dired) (winum) (ob-prolog))
        elpa       (org-plus-contrib ob-prolog memoize android-mode all-the-icons)
        unpin      (n/a)
```
</pre></details>

**Comment by hlissner** [comment]
`.ignore` files should work. With

```
dir/
dir2/
```

In it. Not sure if the leading / is the issue here, but as long as you have ripgrep or fd installed, it will obey ignore files.

**Comment by viocost** [comment]
Not working... Tried also without leading / and with trailing /.

**Comment by hlissner** [comment]
Odd. It works fine for me, in vanilla Doom, using either ripgrep or fd.

What's the value of `projectile-generic-command` and `projectile-git-command`? (Use `SPC h v name-of-variable` to check).

**Comment by viocost** [comment]
projectile-generic-command
```
"rg -0 --files --color=never --hidden --glob '!~/.emacs.d/.local/' --glob '!.idea' --glob '!.ensime_cache' --glob '!.eunit' --glob '!.git' --glob '!.hg' --glob '!.fslckout' --glob '!_FOSSIL_' --glob '!.bzr' --glob '!_darcs' --glob '!.tox' --glob '!.svn' --glob '!.stack-work'"
```

projectile-git-command
```
"rg -0 --files --color=never --hidden --glob '!~/.emacs.d/.local/' --glob '!.idea' --glob '!.ensime_cache' --glob '!.eunit' --glob '!.git' --glob '!.hg' --glob '!.fslckout' --glob '!_FOSSIL_' --glob '!.bzr' --glob '!_darcs' --glob '!.tox' --glob '!.svn' --glob '!.stack-work'"

```

**Comment by hlissner** [comment]
Oh, perhaps what you're seeing is the cache. Press `SPC p i` or `SPC u SPC SPC` to refresh the cache.

**Comment by viocost** [comment]
> Oh, perhaps what you're seeing is the cache. Press `SPC p i` or `SPC u SPC SPC` to refresh the cache.

Thank you Henrik, that was it. I've realized that ripgrep was not installed after all, I was confused by presence of projectile-ripgrep function assuming that ripgrep is also installed.

I've installed ripgrep, then invalidated project cache and re-opened the project. 
My .ignore file looks like this:
```
dir1
dir2
```
Now all is working as expected.

And thanks for developing Doom Emacs, I just love it!
